dispatch/api/serializers.py

```diff
@@ -560,6 +560,8 @@ def update(self, instance, validated_data):
         instance.template = validated_data.get('template_id', instance.template)
         instance.template_data = validated_data.get('template_data', instance.template_data)
 
+        if instance.template == 'timeline':
+            TimelineValidator(validated_data.get('template_data', None))
```

Comment:
Keep this for now, but can you add a TODO?

```py
# TODO: create abstract template validator

# Example:
TemplateValidator(instance.template)
```

I think a more general approach would be to specify required fields and data types in the template definition and then check those in a generic validator. This way any template can be validated in the future.